tracks/go/exercises/isogram/mentoring.md

```diff
@@ -0,0 +1,109 @@
+# Isogram
+
+This exercise further reinforces concepts introduced in previous exercises regarding differences between `rune` and `string`.  
+Students are encouraged to find ways to keep track of unique characters found in the input.
+
+As always, _Unformatted or unlinted code is a reason not to approve._
+
+### Reasonable solutions
+
+```
+package isogram
+
+import "unicode"
+
+// IsIsogram returns true if a word is an isogram
+func IsIsogram(word string) bool {
+	repeated := map[rune]bool{}
+	for _, v := range word {
+		if v == '-' || v == ' ' {
+			continue
+		}
+		v = unicode.ToLower(v)
+		if repeated[v] {
+			return false
+		}
+		repeated[v] = true
+	}
+	return true
+}
+```
+
+### Common suggestions
+
+**Storage**
+
+- A `map[rune]bool` is a straightforward way to store unique runes.  
+  It makes it easier to check for the existence of keys:
+
+```
+if repeated[v] { ... }
+```
+
+instead of using the longer form:
+
+```
+if _, ok := repeated[v]; ok { ... }
+```
+
+- `map[rune]struct{}` uses less memory, but is more awkward to look up; 26 bytes for a map of `bool` is a price worth paying for the improved readability.
+- `map[rune]int` is a reasonable approach where we return false when the count is more than 1, but this is wasteful of memory, and requires a more complicated `if` condition than the simple `if repeated`.
+- Using an array for storage is an approach that will considerably improve the performance of the solution:
+
+```
+var alpha [26]bool
+```
+
+However later on in the code, some variation of the following will need to be adopted:
+
+```
+index := charVal - 'A'
+if alpha[index] {
+	return false
+}
+alpha[index] = true
+```
+
+Working directly with code points this way is both a clever and efficient way to solve the problem, at the expense of readability.  
+If the student goes this route we may suggest the `map[rune]bool` solution instead, and direct them to [Writing Slower Go Programs](https://bitfieldconsulting.com/golang/slower) for tips on when to trade off readability against performance (hint: hardly ever).
```

Comment:
Maybe also add that an index check should be done in this case. If there is a strange character in the string (which shouldn't happen but could happen), it would panic without an index range check.

Reply:
Note: IsLetter does not help here:

```go
func main() {
	fmt.Println('ä', unicode.IsLetter('ä'))
}
// Output: 228 true
```